nbqa/__main__.py

```diff
@@ -270,11 +287,38 @@ def _get_arg(
     if Path(root_dir).is_dir():
         arg = Path(tmpdirname) / Path(root_dir).resolve().relative_to(project_root)
     else:
-        assert len(nb_to_py_mapping) == 1
-        arg = next(iter(nb_to_py_mapping.values()))
+        arg = nb_to_py_mapping[Path(root_dir)]
     return arg
 
 
+def _get_all_args(
+    root_dirs: List[str],
+    tmpdirname: str,
+    nb_to_py_mapping: Dict[Path, Path],
+    project_root: Path,
+) -> List[Path]:
+    """
+    Get all arguments to run command against.
+
+    Parameters
+    ----------
+    root_dirs
+        All notebooks or directories third-party tool is being run against.
+    tmpdirname
+        Temporary directory where converted notebooks are stored.
+    nb_to_py_mapping
+        Mapping between notebooks and Python files corresponding to converted notebooks.
+    project_root
+        Root of repository, where .git / .hg / .nbqa.ini file is.
+
+    Returns
+    -------
+    List[Path]
+        All notebooks or directories to run third-party tool against.
+    """
+    return [_get_arg(i, tmpdirname, nb_to_py_mapping, project_root) for i in root_dirs]
```

Comment:
Minor hint: using enumerators help use memory more efficiently and also helps with CPU usage eventually

Reply:
:thinking: are you sure? I don't see why using an unused enumerator would help

Reply:
Try to apply `enumerate` to other `for` loops as well

Reply:
Enumerators allow lazy evaluation - to see if it helps or not the test benchmarks would help, I could be wrong and also if the number of items in the list are small, it may not be applicable but worth a try.

Reply:
So the benefit of enumerate is not just what you mention but the discussion won't be necessary as the size of the list is too small to take any advantage of these suggestions.